Easy to understand README

black7375 opened the pull request:
- [x] Quick Guide
- [ ] How to Use Refactoring

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
 
+**Modules**
+
+This is [environment](https://github.com/sorin-ionescu/prezto/tree/master/modules/environment), [terminal](https://github.com/sorin-ionescu/prezto/tree/master/modules/terminal) Sample:
```
Maybe add a preface here? I don't really understand what you're going for when you start with "This is ... "

black7375 replied:
Added

> Let's use prezto modules with a real example.

bbenzikry replied:
I meant a preface that explains what you're trying to show, not an added preface while leaving the same sentence. 
For example

Example - Importing the [environment](link) and [terminal]() Prezto modules

black7375 replied:
> Importing the environment and terminal Prezto modules

good :D

bbenzikry commented on README.md:
```diff
@@ -12,31 +12,51 @@
+  - [Install](#install)
+    - [Automatic Installation (Recommended)](#automatic-installation-recommended)
+    - [Manual Installation](#manual-installation)
+  - [Useage](#useage)
```
Usage

bbenzikry commented on README.md:
```diff
@@ -12,31 +12,51 @@
+    - [Load Plugin](#load-plugin)
+    - [Migration](#migration)
+    - [Example Usage](#example-usage)
+- [How to Use](#how-to-use)
```
This is not so clear. You have elements that basically say the same things: 
- Usage
- Example Usage
- How to use

bbenzikry commented on README.md:
```diff
@@ -12,31 +12,51 @@
+    - [Introduction](#introduction)
+    - [Load Plugin](#load-plugin)
+    - [Migration](#migration)
+    - [Example Usage](#example-usage)
```
Example usage points to the wrong link

bbenzikry commented on README.md:
```diff
@@ -701,22 +721,24 @@ Wiki](https://zdharma.github.io/zinit/wiki/). The README is an introductory over
-`zinit.zsh` and setting up completion for command `zinit`. After installing and
-reloading the shell compile Zinit with `zinit self-update`.
+This will install Zinit in `~/.zinit/bin`.
+
```
Extra space

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
 After installing and reloading the shell compile Zinit with `zinit self-update`.
 
-# Usage
+## Useage
```
Usage, also see my above comments about clarity

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
 [Click here to read the introduction to Zinit](https://zdharma.github.io/zinit/wiki/INTRODUCTION/). It explains basic usage and some of the more unique features of Zinit such as the Turbo mode. If you're new to Zinit we highly recommend you read it at least once.
 
-## Example Usage
+### Load Plugin
```
I would suggest something like ### Plugins and snippets

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet https://gist.githubusercontent.com/hightemp/5071909/raw/
+```
+
+### Migration
````
I would suggest "Migrating from other frameworks"

black7375 replied:
Wouldn't it be better to be simple?

black7375 replied:
I'll keep it first

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+
+Then you can use the themes as snippets (`zinit snippet <file path or GitHub URL>`).
+Some themes require not only Oh My Zsh's Git **library**, but also Git **plugin** (error
+about `current_branch` function can be appearing). Load this Git-plugin as single-file
```
error about `current_branch` may appear

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet OMZT::robbyrussell
+```
+
+If the `git` library will not be loaded, then similar to following errors will be appearing:
````
Kind of repetitive, I would just mention that it may cause errors and needs to be loaded and leave it at that. over specificity can also be problematic, especially as other problems may occur because of the plugin itself

black7375 replied:
Oh My Zsh's themes are entirely dependent and can lead to frequent mistakes.
I will make it ***F&A*** item.

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+<details>
+  <summary><b>Migration from Prezto</b></summary>
+
+It's simillar Oh-My-Zsh.
```
I don't think it matters if it's similar or not, I would just link to Prezto

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet PZTM::terminal
+```
+
+F&A: What is `zstyle`?
````
I'm not sure this is the place for it. This should be a README about zinit, but this is more about zsh, and there are a ton of other features in zsh :)

black7375 replied:
It's a migration, but it can also be referenced by novice users when loading prezto's module.
`zstyle` is a very heterogeneous feature when it comes to bash.

I think it doesn't matter if there is no.

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet PZTM::git
+```
+
+Use `zinit ice as"null"` if don't exist `*.plugin.zsh`, `init.zsh`, `*.zsh-theme*` files in module.
````
Not so clear, as null can be used regardless

black7375 replied:
yeah.
However, this part is a tutorial and I think it's not bad for showing how to use it.

bbenzikry replied:
My point is that it doesn't elaborate on why you would use this if those files don't exist

black7375 replied:
I think excessive explanation can be bad in the tutorial.

https://documentation.divio.com/tutorials/#tutorials
> Focus on concrete steps, not abstract concepts
> Provide the minimum necessary explanation

---

Do you have any good phrases?
I don’t think well, :cry:

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+Use `zinit ice atclone"git clone <repo> <location>"` if module have external module.
+Like [completion](https://github.com/sorin-ionescu/prezto/tree/master/modules/completion):
+```shell
+zplugin ice svn blockf \ # blockf: `zinit` manages $fpath. Therefore, $fpath become clean.
````
Comment should be along the lines of the blockf description in the table. 
```
# use blockf to prevent any unnecessary additions to fpath, as zinit manages fpath. 
```

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+- `depth` => `depth`
+</details>
+
+### More Example Usage
```
Pretty confusing title

bbenzikry replied:
I also think the best place for this is where it was, right after the intro. That way people can just copy paste and hack without going through the entire README. This really helped me when I started using zinit

black7375 replied:
I changed `More Example Usage` => `More Examples`
As I wrote below, I want to hear more opinions about the location.

bbenzikry commented on README.md:
```diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+### Migration
+<details>
+  <summary><b>Migration from Oh-My-ZSH</b></summary>
+Just load with Snippet!!
```
No need for this line

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet OMZP::osx
+```
+
+Use `zinit ice as"completion"` if single file is completion.
````
I would suggest 

Use `zinit ice as"completion"` to directly add single file completion snippets

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit snippet OMZP::fd/_fd
+```
+
+See also the Wiki page: [Example Oh My Zsh
````
For more information, visit our [wiki](http://zdharma.github.io/zinit/wiki/Example-Oh-My-Zsh-setup)

black7375 replied:
For more information, visit [Wiki:Example Oh My Zsh Setup](http://zdharma.github.io/zinit/wiki/Example-Oh-My-Zsh-setup/).

bbenzikry replied:
The reasoning I had behind that is that we don't have to change the README if the title changes on the wiki

bbenzikry replied:
At least from a view perspective

black7375 replied:
There are other ways too.

What do you think of the existing linking method?
https://github.com/zdharma/zinit#example-usage
[You can see an extended explanation of LS_COLORS in the Wiki.](https://zdharma.github.io/zinit/wiki/LS_COLORS-explanation/)
[You can see an extended explanation of direnv in the Wiki.](https://zdharma.github.io/zinit/wiki/Direnv-explanation/)


=>
[You can see an extended explanation of Oh-My-Zsh setup in the Wiki](http://zdharma.github.io/zinit/wiki/Example-Oh-My-Zsh-setup/)

bbenzikry replied:
I don't mind, they're both OK

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zplugin snippet PZTM::completion
+```
+
+**Prompt(Theme)**
````
Why is this under Prezto? Seems it makes more sense to give a prompt example for zinit as a whole

bbenzikry replied:
Just a suggestion - we can also add [starship](https://starship.rs) to the prompt list, it's extremely popular.
You can see an example of what I do in my dotfiles
[here](https://github.com/bbenzikry/dotfiles/blob/macos/.config/zsh/loaders/02-prompt.zsh)

black7375 replied:
Compilation is burdensome for general users.
Let's use the release at example.

black7375 replied:
I'll use this.

```zsh
# Load starship theme
zinit ice as"command" from"gh-r" \
          atclone"./starship init zsh > init.zsh; ./starship completions zsh > _starship" \
          atpull"%atclone" src"init.zsh"
zinit light starship/starship
```

bbenzikry replied:
> Compilation is burdensome for general users.
> Let's use the release at example.

I didn't really understand - I was talking about prompt addition in general

black7375 replied:
> I didn't really understand - I was talking about prompt addition in general

Ah..I looked at your settings.
```zsh
#█▓▒░ Starship prompt, neofetch, hacker-quotes
zinit as"null" lucid for \
    has"cargo" ver"master" atclone'cargo build --release' atpull'%atclone' lbin'!target/release/starship -> starship' atload'!starship_standalone' \
    starship/starship \
    has"starship" as"completion" id-as"starship_completions" atclone"starship completions zsh > _starship" zdharma/null
```

If someone are not a Rust user, they may not have `cargo`, and compile time and space requirements may be an issue.
So I configured the example in a way that only downloads binary from github release.
I just wanted to talk about the prompt setup code that will be used in the documentation.

Sorry if you are offended.

bbenzikry replied:
Ah, I see - yeah, I didn't mean for that to be copied step by step - I'm using cargo as I actually worked a bit on the starship repo and wanted to have master built.

In any case what I meant was that the prompt heading is under Prezto and I think it should be a general place to show people how to configure a prompt with zinit, regardless of what framework they are migrating from.

black7375 replied:
It was a big mistake.
I got it right now 😄 

https://github.com/black7375/zinit#plugins-and-snippets

bbenzikry commented on README.md:
```diff
@@ -12,31 +12,51 @@
-- [Usage](#usage)
-  - [Introduction](#introduction)
-  - [Example Usage](#example-usage)
+- [Quick Start](#quick-start)
```
After reviewing all of it, I suggest the following structure instead
- Quick start
-- Install
-- Example ( with the large zsh snippet )
- Usage
-- Introduction
-- Loading plugins
-- Migrating from other frameworks
-- Ice modifiers 
... the rest

black7375 replied:
Motive for this PR plan is the large zsh snippet.
It may be practical, but it's too difficult for newbies.
The impression I got was that it should appear after finishing the tutorial.

I want to hear from beginners.

bbenzikry replied:
Well I think we should get some decision going as it'll take a while for zinit beginners to go straight into suggesting something in this PR ;]
I will say that when I started out with zinit ( so that qualifies as a beginner I guess ), it was really easy for me to just dive in ( as someone who was familiar with zsh ) using the "in your face" example

black7375 replied:
To speed things up, I'll also ask for help on Reddit.

bbenzikry reviewed (changes requested):
Hi @black7375, thanks for working on this - I've reviewed what I can. some of my comments are on previous terminology from the old README and some touch on your new efforts.

black7375 commented:
@bbenzikry  I forgot it. lol
I will reflect on your opinion and work on it again tomorrow.

Thanks for reminding.

bbenzikry commented:
:smile: Sorry for dropping in out of nowhere

black7375 commented:
Your review was very helpful. :+1:

bbenzikry commented on README.md:
````diff
@@ -741,24 +763,338 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+zinit light <repo/plugin> # Load without reporting/investigating.
+```
+
+If you want source local or remote file (by direct URL) using `snippet`.
````
What is the full sentence? 
In general you can do so seems to be more proper

black7375 replied:
It's paragraph

---

If you want source local or remote files (using direct URL), you can do with `snippet`.
```
zinit snippet <URL>
```

bbenzikry replied:
Sorry, I just feel it's bad english without "to" and "so"
If you want **to** source 
you can do **so** with

black7375 replied:
Ok, I'll fix it.

black7375 commented on README.md:
````diff
@@ -741,24 +762,350 @@ Various paths can be customized, see section [Customizing Paths](#customizing-pa
+Like [completion](https://github.com/sorin-ionescu/prezto/tree/master/modules/completion):
+```shell
+zplugin ice svn blockf \ # use blockf to prevent any unnecessary additions to fpath, as zinit manages fpath
+atclone"git clone --recursive https://github.com/zsh-users/zsh-completions.git external"
````
Indent.
I'll fix next commit.

black7375 commented:
**2021/04/21 update**

Added content about turbo mode and lucid options.
https://github.com/black7375/zinit#turbo-and-lucid

---

[reddit post](https://www.reddit.com/r/zsh/comments/mur6eu/anyone_interested_in_zinit_documentation/)

>  I consider myself a pretty advanced Zsh aficionado and I could never make heads or tails of Zplugin/Zinit documentation. I’ve used nearly every other Zsh plugin manager but not zinit solely because the docs are so inaccessible and introduce so many concepts and keywords that make absolutely no sense. What does “light” do? Why do I need “ice”?

It's a similar point to #460
I added F&A for `ice`.

> I'd really appreciate it if you could add a list where users can submit examples of how to load a specific plugin. I've found most of what I needed by searching Github issues, but often the information is out of date, or without any explanations.

I'm thinking about adding real world examples.
- [zdharma/zinit configs](https://github.com/zdharma/zinit-configs)
- [black7375/BlaCk-Void-Zsh](https://github.com/black7375/BlaCk-Void-Zsh): However, I am cautious because it can be viewed as a promotion of my project.

pschmitt commented:
I think that the best way forward in regards to an up to date list of zinit ices for popular plugins would be to use GitHub's wiki.

[zinit packages](https://zdharma.github.io/zinit/wiki/Zinit-Packages/) are truly the best technical solution for this problem but there isn't much traction there.

black7375 commented:
Wiki also needs improvements.
The code style is inconsistent.
(load/light, as"command"/as"program", ..etc)

I've never used zinit packages 😢

bbenzikry commented:
> I think that the best way forward in regards to an up to date list of zinit ices for popular plugins would be to use GitHub's wiki.
> 
> [zinit packages](https://zdharma.github.io/zinit/wiki/Zinit-Packages/) are truly the best technical solution for this problem but there isn't much traction there.

I totally agree - I would even take it a step further - @pschmitt  @psprint do you think a GH wiki can substitute the current wiki? 
I would be happy to help with work there.

black7375 commented:
> do you think a GH wiki can substitute the current wiki?

I think it's possible enough

I will follow @psprint 's opinion

luxus commented:
oh wow.. thats a lot work black7275
I wonder if you can make a example how https://github.com/zinit-zsh/z-a-meta-plugins works
for newbies this would help to install common plugins without hundred lines

black7375 commented:
@luxus 
I've never used z-a-meta-plugins, so I'm not sure.

@psprint 
I'm running out of time because I'm working on something else for a while.
Do you want to merge now?
If not, you probably need more time.

deadmanworking commented:
@black7375 This has been very helpful. Some thoughts. 

1. Expanding the wiki to have articles where there can be specific pages for using Zinit to do things. (Ex. I wanted to poke around with anaconda and I found a reference [via google] in the Zinit issues that helped me sort it out. ) I've found that the subreddit and google can yield inconsistent results. It would be nice to be able to go to this repo, look in the wiki for that thing I want to automate for every computer I touch (I think that is why we do this, right?) and also be able to understand why it works like it does. (I say this with full intention of contributing if a thing was an actual thing) 

2. The biggest thing I see missing from the documentation is the 'why'. A lot of people (myself included) are not experts with shell scripting and/or zsh. With things like macOS defaulting to zsh, there are a lot of people who could benefit from zinit but are likely intimidated by the learning curve.) 

3. All of the above aside, I've been stuck on some things and this README helped me sort it out. I'm game to help sort things out if @black7375 is busy doing important other things.

black7375 commented:
Yes, Barriers to entry are significant for the general people.
`zinit` is quite complex, so I focused on the usage itself.

If you send PR, I will be grateful.
I asked @psprint  last time(when no conflict), but he doesn't seem to have a plan to merge yet.

psprint commented:
Thanks. Is there anything that needs to be done ? I would like to merge.

xenoterracide commented:
@psprint totally not my call but it looks like you have a conflict that needs to be resolved before you could ;)

black7375 commented:
I'll try to resolve the conflict soon and add `blockf`, plugin update related content.

black7375 commented:
I'am ready.
Thanks for making such a great program.

psprint commented:
Thanks!
